src/client.js

```diff
@@ -23,70 +23,95 @@ const API_TYPE_CLIENT = 'client';
+          [API_TYPE_CLIENT]: `${baseUrl}/api/client/v1.0`,
+          [API_TYPE_PRIVATE]: `${baseUrl}/api/private/v1.0`,
+          [API_TYPE_APP]: (clientAppID ?
+            `${baseUrl}/api/client/v1.0/app/${clientAppID}` :
+            `${baseUrl}/api/public/v1.0`)
+        },
+        [v2]: {
+          [API_TYPE_PUBLIC]: `${baseUrl}/api/public/v2.0`,
+          [API_TYPE_CLIENT]: `${baseUrl}/api/client/v2.0`,
+          [API_TYPE_PRIVATE]: `${baseUrl}/api/private/v2.0`,
+          [API_TYPE_APP]: (clientAppID ?
+            `${baseUrl}/api/client/v2.0/app/${clientAppID}` :
+            `${baseUrl}/api/public/v2.0`)
+        },
+        [v3]: {
+          [API_TYPE_PUBLIC]: `${baseUrl}/api/public/v3.0`,
+          [API_TYPE_CLIENT]: `${baseUrl}/api/client/v3.0`,
+          [API_TYPE_APP]: (clientAppID ?
+            `${baseUrl}/api/client/v3.0/app/${clientAppID}` :
+            `${baseUrl}/api/admin/v3.0`)
+        }
+      };
 
-    this.authUrl = (
-      clientAppID ?
-        `${baseUrl}/api/client/v2.0/app/${clientAppID}/auth` :
-        `${baseUrl}/api/admin/v3.0/auth`
-    );
+      const authOptions = {
+        codec: APP_CLIENT_CODEC,
+        storage: options.storage
+      };
 
-    this.rootURLsByAPIVersion = {
-      [v1]: {
-        [API_TYPE_PUBLIC]: `${baseUrl}/api/public/v1.0`,
-        [API_TYPE_CLIENT]: `${baseUrl}/api/client/v1.0`,
-        [API_TYPE_PRIVATE]: `${baseUrl}/api/private/v1.0`,
-        [API_TYPE_APP]: (clientAppID ?
-          `${baseUrl}/api/client/v1.0/app/${clientAppID}` :
-          `${baseUrl}/api/public/v1.0`)
-      },
-      [v2]: {
-        [API_TYPE_PUBLIC]: `${baseUrl}/api/public/v2.0`,
-        [API_TYPE_CLIENT]: `${baseUrl}/api/client/v2.0`,
-        [API_TYPE_PRIVATE]: `${baseUrl}/api/private/v2.0`,
-        [API_TYPE_APP]: (clientAppID ?
-          `${baseUrl}/api/client/v2.0/app/${clientAppID}` :
-          `${baseUrl}/api/public/v2.0`)
-      },
-      [v3]: {
-        [API_TYPE_PUBLIC]: `${baseUrl}/api/public/v3.0`,
-        [API_TYPE_CLIENT]: `${baseUrl}/api/client/v3.0`,
-        [API_TYPE_APP]: (clientAppID ?
-          `${baseUrl}/api/client/v3.0/app/${clientAppID}` :
-          `${baseUrl}/api/admin/v3.0`)
+      if (options.storageType) {
+        authOptions.storageType = options.storageType;
+      }
+      if (options.platform) {
+        authOptions.platform = options.platform;
+      }
+      if (options.authCodec) {
+        authOptions.codec = options.authCodec;
       }
-    };
-
-    const authOptions = {
-      codec: APP_CLIENT_CODEC,
-      storage: options.storage
-    };
 
-    if (options.storageType) {
-      authOptions.storageType = options.storageType;
-    }
-    if (options.platform) {
-      authOptions.platform = options.platform;
-    }
-    if (options.authCodec) {
-      authOptions.codec = options.authCodec;
-    }
+      const authFactory = AuthFactory.create(this, this.authUrl, authOptions);
+      authFactory.then(auth => {
```

Comment:
With my suggestion, this would likely become just a return since we are outside of constructor.